File: src/WorkflowShell/mvn/MavenPomChange.rb
require_relative '../models/Command'

class MvnPomChange < Command
  @command_string
  @command_description
  @command_usage

  def initialize
    @command_string = 'mvnpomchange'
    @command_description = 'Changes the version of the module in the current directory and all submodules to the input version.'
    @command_usage = @command_string + " <New Version>"
  end

  def run_command(arguments)
    # Ask the Command class to parse the basic options like help and verbosity
    parser_components = parse_options(arguments, false)

    new_version = parser_components.leftover_arguments.pop
    if new_version.nil?
      puts "You MUST specify the new version!"
      puts ""
      puts "Current version is..."
      run_shell_command("mvn help:evaluate -Dexpression=project.version|grep -Ev '(^\\[|Download\\w+:)'", false)
      print_help(parser_components.basic_options_printer, nil)
    end

    command = "mvn versions:set -DnewVersion=#{new_version} -Djava.awt.headless=true -DgenerateBackupPoms=false"
    run_shell_command(command, parser_components.basic_options.verbose)
  end
end

File: src/WorkflowShell/mvn/MavenPomChange.js
const Command = require('../models/Command');

class MavenPomChange extends Command {

  static getString() {
    return 'mpomchange';
  }

  static getDescription() {
    return 'Changes the version of the module in the current directory and all submodules to the input version.';
  }

  getUsage() {
    return `${MavenPomChange.getString()} <New Version>`;
  }

  run(args) {
    const ok = super.run(args);
    if (!ok) {
      return false;
    }

    const newVersion = args._.shift();
    if (!newVersion) {
      console.log('You MUST specify the new version!');
      console.log('');
      console.log('Current version is...');
      super.execute("mvn help:evaluate -Dexpression=project.version|grep -Ev '(^\\[|Download\\w+:)'");
      this.printHelp();
      return false;
    }

    const baseCommand = `mvn versions:set -DnewVersion=${newVersion} -Djava.awt.headless=true -DgenerateBackupPoms=false`;
    return super.execute(baseCommand);
  }
}

module.exports = MavenPomChange;
